js/src/components/adaptive-form/adaptive-form-context.js

```diff
@@ -0,0 +1,50 @@
+/**
+ * External dependencies
+ */
+import { createContext, useContext } from '@wordpress/element';
+
+/**
+ * @typedef {Object} InputProps
+ * @property {*} value Form value.
+ * @property {boolean} checked Form value converted to boolean.
+ * @property {*} selected Form value.
+ * @property {(value: Event<HTMLInputElement> | *) => void} onChange Function to handle onChange event.
+ * @property {() => void} onBlur Function to handle onBlur event.
+ * @property {'has-error' | undefined} className 'has-error' if the form value is invalid and marked as touched. `undefined` otherwise.
+ * @property {string | undefined | null} help The corresponding value in form `errors` if the form value is marked as touched. `null` otherwise.
+ */
+
+/**
+ * @typedef {Object} AdaptiveFormContextAdapter
+ * @property {boolean} isSubmitting `true` if the form is currently being submitted.
+ * @property {boolean} isSubmitted Set to `true` after the form is submitted. Initial value and during submission are set to `false`.
+ * @property { HTMLElement | null} submitter Set to the element triggering the `handleSubmit` callback until the processing of `onSubmit` is completed. `null` otherwise.
+ */
+
+/**
+ * @typedef {Object} AdaptiveFormContext
+ * @property {Object} values Form values.
+ * @property {Object} errors Object with key-value pairs representing errors for form values. Empty object if no errors.
+ * @property {Object} touched Object with key-value pairs representing form values is touched.
+ * @property {boolean} isValidForm `true` if form values pass the validation.
+ * @property {boolean} isDirty `true` after any of the form values is modified.
+ * @property {(name: string, value: *) => void} setValue Function to set a form value.
+ * @property {(name: string) => InputProps} getInputProps Function to get the corresponding input props by the name of a form value.
```

Comment:
The part of `by the name of a form value` seems unclear for me. As far as I understand we have a set of form inputs with props (let's say name, and age). So if we do  `getInputProps('name')` we are getting the props for the `name` input which has a value inside among other props. So i would rephrase this as. 

`Function to get the corresponding InputProps object using the input name as a key.`

Reply:
Using "input name" might be confusing since the `name` should be one of the names of the given form `values`. I rephrased it in https://github.com/woocommerce/google-listings-and-ads/pull/1794/commits/c9f90ec2192e4c72e93e55059fb15c6a2296548a. Is this clear now?